structure/deque/deque.go

```diff
@@ -0,0 +1,99 @@
+/*
+Package deque implements a Double Ended Queue data structure.
+
+Description:
+	Double Ended Queue is a generalized version of Queue data structure
+	that allows insert and delete at both ends.
+
+Basic Operations:
+	enQueueFront(): Adds an item at the front of Deque.
+	enQueueRear(): Adds an item at the rear of Deque.
+	deQueueFront(): Deletes an item from front of Deque.
+	deQueueRear(): Deletes an item from rear of Deque.
+	getFront(): Gets the front item from queue.
+	getRear(): Gets the last item from queue.
+	isEmpty(): Checks whether Deque is empty or not.
+	getLength(): Gets the length of Deque.
+
+References:
+	Wikipedia : https://en.wikipedia.org/wiki/Double-ended_queue
+	Github: https://www.geeksforgeeks.org/deque-set-1-introduction-applications/
+
+Author: [Sayan](https://github.com/bose-sayan)
+*/
```

Comment:
This comment style would not be detected by any of the go based documentation tools. Please read the guidance on how to document go code [here](https://go.dev/blog/godoc) 🙏🏼

Reply:
Hey! Thanks for the clarification. I refactored the comments accordingly, but am not sure if the style aligns with how go doc expects it to be. It would be really great if you could check it out once. Thank you again!